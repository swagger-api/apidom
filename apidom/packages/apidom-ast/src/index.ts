// JSON AST related exports
export { default as JsonNode } from './json/nodes/JsonNode';
export { default as JsonDocument } from './json/nodes/JsonDocument';
export { default as JsonObject } from './json/nodes/JsonObject';
export { default as JsonProperty } from './json/nodes/JsonProperty';
export { default as JsonArray } from './json/nodes/JsonArray';
export { default as JsonValue } from './json/nodes/JsonValue';
export { default as JsonKey } from './json/nodes/JsonKey';
export { default as JsonString } from './json/nodes/JsonString';
export { default as JsonStringContent } from './json/nodes/JsonStringContent';
export { default as JsonEscapeSequence } from './json/nodes/JsonEscapeSequence';
export { default as JsonNumber } from './json/nodes/JsonNumber';
export { default as JsonTrue } from './json/nodes/JsonTrue';
export { default as JsonFalse } from './json/nodes/JsonFalse';
export { default as JsonNull } from './json/nodes/JsonNull';
export {
  isFalse as isJsonFalse,
  isProperty as isJsonProperty,
  isStringContent as isJsonStringContent,
  isEscapeSequence as isJsonEscapeSequence,
  isArray as isJsonArray,
  isKey as isJsonKey,
  isNull as isJsonNull,
  isNumber as isJsonNumber,
  isObject as isJsonObject,
  isString as isJsonString,
  isTrue as isJsonTrue,
} from './json/nodes/predicates';
// YAML AST related exports
export { default as YamlAlias } from './yaml/nodes/YamlAlias';
export { default as YamlCollection } from './yaml/nodes/YamlCollection';
export { default as YamlComment } from './yaml/nodes/YamlComment';
export { default as YamlDirective } from './yaml/nodes/YamlDirective';
export { default as YamlDocument } from './yaml/nodes/YamlDocument';
export { default as YamlKeyValuePair } from './yaml/nodes/YamlKeyValuePair';
export { default as YamlMapping } from './yaml/nodes/YamlMapping';
export { default as YamlNode } from './yaml/nodes/YamlNode';
export { default as YamlScalar } from './yaml/nodes/YamlScalar';
export { default as YamlSequence } from './yaml/nodes/YamlSequence';
export { default as YamlStream } from './yaml/nodes/YamlStream';
export { default as YamlTag } from './yaml/nodes/YamlTag';
export { default as YamlAnchor } from './yaml/nodes/YamlAnchor';
export {
  isAlias as isYamlAlias,
  isKeyValuePair as isYamlKeyValuePair,
  isDirective as isYamlDirective,
  isDocument as isYamlDocument,
  isMapping as isYamlMapping,
  isScalar as isYamlScalar,
  isSequence as isYamlSequence,
  isStream as isYamlStream,
  isTag as isYamlTag,
} from './yaml/nodes/predicates';
// generic AST related exports
export { default as Literal } from './Literal';
export { Point, default as Position } from './Position';
export { default as Error } from './Error';
export { default as ParseResult } from './ParseResult';
// AST traversal related exports
export { getVisitFn, BREAK, visit } from './visitor';
// YAML CST/AST transformers related exports
export {
  transform as transformTreeSitterYamlCST,
  keyMap as treeSitterYamlKeyMap,
} from './yaml/transformers/tree-sitter-yaml';
