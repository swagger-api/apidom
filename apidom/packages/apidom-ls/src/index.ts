// @ts-ignore
import ApiDOMParser from 'apidom-parser';
// @ts-ignore
import * as openapi3_1Adapter from 'apidom-parser-adapter-openapi-json-3-1';

export {
  isRefElement,
  isLinkElement,
  isMemberElement,
  isObjectElement,
  isArrayElement,
  isBooleanElement,
  isNullElement,
  isElement,
  isNumberElement,
  isStringElement,
} from 'apidom';

export { default as getLanguageService } from './apidom-language-service';

export type {
  LanguageService,
  LanguageSettings,
  SymbolsContext,
  ValidationContext,
  CompletionContext,
  WorkspaceContextService,
  ColorsContext,
  LanguageServiceContext,
  SUPPORTED_LANGUAGES,
} from './apidom-language-types';

export const pocTODOToRemove: () => Promise<string> = async () => {
  const parser = ApiDOMParser();

  const value = `{
    "openapi": "3.0.0",
      "info": {
        "version": "0.1.9"
      }
    }`;
  parser.use(openapi3_1Adapter);
  // parser.use(asyncapi2_0Adapter);
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const parseResult = await parser.parse(value, { sourceMap: true });
  // console.log('Par', JSON.stringify(parseResult));
  return 'done';
};
