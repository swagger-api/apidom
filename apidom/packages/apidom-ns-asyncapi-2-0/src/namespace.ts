import { NamespacePluginOptions } from 'minim';

/**
 * AsyncApi 2.0.0 specification elements.
 */
import AsyncApi2_0Element from './elements/AsyncApi2-0';
import AsyncApiVersionElement from './elements/AsyncApiVersion';
import ChannelBindingsElement from './elements/ChannelBindings';
import ChannelItemElement from './elements/ChannelItem';
import ChannelsElement from './elements/Channels';
import ComponentsElement from './elements/Components';
import ContactElement from './elements/Contact';
import CorrelationIDElement from './elements/CorrelationID';
import DefaultContentTypeElement from './elements/DefaultContentType';
import ExternalDocumentationElement from './elements/ExternalDocumentation';
import IdentifierElement from './elements/Identifier';
import InfoElement from './elements/Info';
import LicenseElement from './elements/License';
import OAuthFlowElement from './elements/OAuthFlow';
import OAuthFlowsElement from './elements/OAuthFlows';
import OperationElement from './elements/Operation';
import ParameterElement from './elements/Parameter';
import ParametersElement from './elements/Parameters';
import ReferenceElement from './elements/Reference';
import SchemaElement from './elements/Schema';
import SecurityRequirementElement from './elements/SecurityRequirement';
import ServerElement from './elements/Server';
import ServerBindingsElement from './elements/ServerBindings';
import ServersElement from './elements/Servers';
import ServerVariableElement from './elements/ServerVariable';
import TagElement from './elements/Tag';
import TagsElement from './elements/Tags';
/**
 * Binding elements.
 */
// AMQP
import AmqpChannelBinding0_9Element from './elements/bindings/amqp-0-9/AmqpChannelBinding0-9';
import AmqpMessageBinding0_9Element from './elements/bindings/amqp-0-9/AmqpMessageBinding0-9';
import AmqpOperationBinding0_9Element from './elements/bindings/amqp-0-9/AmqpOperationBinding0-9';
// HTTP
import HttpMessageBindingElement from './elements/bindings/http/HttpMessageBinding';
import HttpOperationBindingElement from './elements/bindings/http/HttpOperationBinding';
// Kafka
import KafkaMessageBindingElement from './elements/bindings/kafka/KafkaMessageBinding';
import KafkaOperationBindingElement from './elements/bindings/kafka/KafkaOperationBinding';
// MQTT
import MqttMessageBindingElement from './elements/bindings/mqtt/MqttMessageBinding';
import MqttOperationBindingElement from './elements/bindings/mqtt/MqttOperationBinding';
import MqttServerBindingElement from './elements/bindings/mqtt/MqttServerBinding';
// WebSocket
import WebSocketChannelBindingElement from './elements/bindings/ws/WebSocketChannelBinding';

const asyncApi2_0 = {
  namespace: (options: NamespacePluginOptions) => {
    const { base } = options;

    /**
     * AsyncApi 2.0.0 specification elements.
     */
    base.register('asyncApi2_0', AsyncApi2_0Element);
    base.register('asyncApiVersion', AsyncApiVersionElement);
    base.register('channelBindings', ChannelBindingsElement);
    base.register('channelItem', ChannelItemElement);
    base.register('channels', ChannelsElement);
    base.register('components', ComponentsElement);
    base.register('contact', ContactElement);
    base.register('correlationID', CorrelationIDElement);
    base.register('defaultContentType', DefaultContentTypeElement);
    base.register('externalDocumentation', ExternalDocumentationElement);
    base.register('identifier', IdentifierElement);
    base.register('info', InfoElement);
    base.register('license', LicenseElement);
    base.register('oAuthFlow', OAuthFlowElement);
    base.register('oAuthFlows', OAuthFlowsElement);
    base.register('operation', OperationElement);
    base.register('parameter', ParameterElement);
    base.register('parameters', ParametersElement);
    base.register('reference', ReferenceElement);
    base.register('schema', SchemaElement);
    base.register('securityRequirement', SecurityRequirementElement);
    base.register('server', ServerElement);
    base.register('serverBindings', ServerBindingsElement);
    base.register('servers', ServersElement);
    base.register('serverVariable', ServerVariableElement);
    base.register('tag', TagElement);
    base.register('tags', TagsElement);

    /**
     * Binding elements.
     */
    // AMQP
    base.register('amqpChannelBindingElement', AmqpChannelBinding0_9Element);
    base.register('amqpMessageBindingElement', AmqpMessageBinding0_9Element);
    base.register('amqpOperationBindingElement', AmqpOperationBinding0_9Element);
    // HTTP
    base.register('httpMessageBindingElement', HttpMessageBindingElement);
    base.register('httpOperationBindingElement', HttpOperationBindingElement);
    // Kafka
    base.register('kafkaMessageBindingElement', KafkaMessageBindingElement);
    base.register('kafkaOperationBindingElement', KafkaOperationBindingElement);
    // MQTT
    base.register('mqttMessageBindingElement', MqttMessageBindingElement);
    base.register('mqttOperationBindingElement', MqttOperationBindingElement);
    base.register('mqttServerBindingElement', MqttServerBindingElement);
    // WebSocket
    base.register('webSocketChannelBindingElement', WebSocketChannelBindingElement);

    return base;
  },
};

export default asyncApi2_0;
