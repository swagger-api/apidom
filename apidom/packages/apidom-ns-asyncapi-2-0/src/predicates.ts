import { allPass, either, is } from 'ramda';
import { createPredicate } from 'apidom';

import AsyncApi2_0Element from './elements/AsyncApi2-0';
import AsyncApiVersionElement from './elements/AsyncApiVersion';
import ChannelBindingsElement from './elements/ChannelBindings';
import ChannelItemElement from './elements/ChannelItem';
import ChannelsElement from './elements/Channels';
import ComponentsElement from './elements/Components';
import ContactElement from './elements/Contact';
import IdentifierElement from './elements/Identifier';
import InfoElement from './elements/Info';
import LicenseElement from './elements/License';
import OperationElement from './elements/Operation';
import ParameterElement from './elements/Parameter';
import ParametersElement from './elements/Parameters';
import ReferenceElement from './elements/Reference';
import SchemaElement from './elements/Schema';
import SecurityRequirementElement from './elements/SecurityRequirement';
import ServerElement from './elements/Server';
import ServerBindingsElement from './elements/ServerBindings';
import ServersElement from './elements/Servers';
import ServerVariableElement from './elements/ServerVariable';

export const isAsyncApi2_0Element = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq, hasClass }) => {
    const isElementTypeAsyncApi2_0 = isElementType('asyncApi2-0');
    const primitiveEqObject = primitiveEq('object');
    const hasClassApi = hasClass('api');

    return either(
      is(AsyncApi2_0Element),
      allPass([hasBasicElementProps, isElementTypeAsyncApi2_0, primitiveEqObject, hasClassApi]),
    );
  },
);

export const isAsyncApiVersionElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeAsyncApiVersion = isElementType('asyncApiVersion');
    const primitiveEqString = primitiveEq('string');

    return either(
      is(AsyncApiVersionElement),
      allPass([hasBasicElementProps, isElementTypeAsyncApiVersion, primitiveEqString]),
    );
  },
);

export const isChannelBindingsElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeChannelBindings = isElementType('channelBindings');
    const primitiveEqString = primitiveEq('object');

    return either(
      is(ChannelBindingsElement),
      allPass([hasBasicElementProps, isElementTypeChannelBindings, primitiveEqString]),
    );
  },
);

export const isChannelItemElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeChannelItem = isElementType('channelItem');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(ChannelItemElement),
      allPass([hasBasicElementProps, isElementTypeChannelItem, primitiveEqObject]),
    );
  },
);

export const isChannelsElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeChannels = isElementType('channels');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(ChannelsElement),
      allPass([hasBasicElementProps, isElementTypeChannels, primitiveEqObject]),
    );
  },
);

export const isComponentsElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeComponents = isElementType('components');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(ComponentsElement),
      allPass([hasBasicElementProps, isElementTypeComponents, primitiveEqObject]),
    );
  },
);

export const isContactElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeContact = isElementType('contact');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(ContactElement),
      allPass([hasBasicElementProps, isElementTypeContact, primitiveEqObject]),
    );
  },
);

export const isIdentifierElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeIdentifier = isElementType('identifier');
    const primitiveEqString = primitiveEq('string');

    return either(
      is(IdentifierElement),
      allPass([hasBasicElementProps, isElementTypeIdentifier, primitiveEqString]),
    );
  },
);

export const isInfoElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeInfo = isElementType('info');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(InfoElement),
      allPass([hasBasicElementProps, isElementTypeInfo, primitiveEqObject]),
    );
  },
);

export const isLicenseElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeLicense = isElementType('license');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(LicenseElement),
      allPass([hasBasicElementProps, isElementTypeLicense, primitiveEqObject]),
    );
  },
);

export const isOperationElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeOperation = isElementType('operation');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(OperationElement),
      allPass([hasBasicElementProps, isElementTypeOperation, primitiveEqObject]),
    );
  },
);

export const isParameterElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeParameter = isElementType('parameter');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(ParameterElement),
      allPass([hasBasicElementProps, isElementTypeParameter, primitiveEqObject]),
    );
  },
);

export const isParametersElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeParameters = isElementType('parameters');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(ParametersElement),
      allPass([hasBasicElementProps, isElementTypeParameters, primitiveEqObject]),
    );
  },
);

export const isReferenceElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeReference = isElementType('reference');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(ReferenceElement),
      allPass([hasBasicElementProps, isElementTypeReference, primitiveEqObject]),
    );
  },
);

export const isSchemaElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeSchema = isElementType('schema');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(SchemaElement),
      allPass([hasBasicElementProps, isElementTypeSchema, primitiveEqObject]),
    );
  },
);

export const isSecurityRequirementElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeSecurityRequirement = isElementType('securityRequirement');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(SecurityRequirementElement),
      allPass([hasBasicElementProps, isElementTypeSecurityRequirement, primitiveEqObject]),
    );
  },
);

export const isServerElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeServer = isElementType('server');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(ServerElement),
      allPass([hasBasicElementProps, isElementTypeServer, primitiveEqObject]),
    );
  },
);

export const isServerBindingsElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeServerBindings = isElementType('serverBindings');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(ServerBindingsElement),
      allPass([hasBasicElementProps, isElementTypeServerBindings, primitiveEqObject]),
    );
  },
);

export const isServersElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeServers = isElementType('servers');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(ServersElement),
      allPass([hasBasicElementProps, isElementTypeServers, primitiveEqObject]),
    );
  },
);

export const isServerVariableElement = createPredicate(
  ({ hasBasicElementProps, isElementType, primitiveEq }) => {
    const isElementTypeServerVariable = isElementType('serverVariable');
    const primitiveEqObject = primitiveEq('object');

    return either(
      is(ServerVariableElement),
      allPass([hasBasicElementProps, isElementTypeServerVariable, primitiveEqObject]),
    );
  },
);
