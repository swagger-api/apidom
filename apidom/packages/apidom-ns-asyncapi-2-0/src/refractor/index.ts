import { refract as baseRefract } from 'minim';
import { Element, dereference } from 'apidom';
import { invokeArgs } from 'ramda-adjunct';

import { visit } from '../traversal/visitor';
import specification from './specification';

const refract = <T extends Element>(
  value: any,
  { specPath = ['visitors', 'document', 'objects', 'AsyncApi', '$visitor'] } = {},
): T => {
  const element = baseRefract(value);
  const resolvedSpec = dereference(specification);
  const visitor = invokeArgs(specPath, [], resolvedSpec);

  // @ts-ignore
  visit(element, visitor, { state: { specObj: resolvedSpec } });

  return visitor.element;
};

export const createRefractor = (specPath: string[]) => (value: any, options = {}) =>
  refract(value, { specPath, ...options });

export default refract;
