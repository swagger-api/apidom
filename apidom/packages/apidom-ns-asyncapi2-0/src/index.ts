export { default } from './namespace';

export {
  isRefElement,
  isLinkElement,
  isMemberElement,
  isObjectElement,
  isArrayElement,
  isBooleanElement,
  isNullElement,
  isElement,
  isNumberElement,
  isStringElement,
} from 'apidom';

export {
  isAsycApi2_0Element,
  isAsycapiElement,
  isComponentsElement,
  isContactElement,
  isIdentifierElement,
  isInfoElement,
  isLicenseElement,
  isSchemaElement,
  isChannelsElement,
  isChannelItemElement,
  isServersElement,
  isServerElement,
} from './predicates';

export { default as AsyncapiElement } from './elements/Asyncapi';
export { default as AsyncApi2_0Element } from './elements/AsyncApi2-0';
export { default as ComponentsElement } from './elements/Components';
export { default as ContactElement } from './elements/Contact';
export { default as IdentifierElement } from './elements/Identifier';
export { default as InfoElement } from './elements/Info';
export { default as LicenseElement } from './elements/License';
export { default as SchemaElement } from './elements/Schema';
export { default as ChannelsElement } from './elements/Channels';
export { default as ChannelItemElement } from './elements/ChannelItem';
export { default as OperationElement } from './elements/Operation';
export { default as ParametersElement } from './elements/Parameters';
export { default as ChannelBindingsElement } from './elements/ChannelBindings';
export { default as ServersElement } from './elements/Servers';
export { default as ServerElement } from './elements/Server';
export { default as ServerVariableElement } from './elements/ServerVariable';
export { default as SecurityRequirementElement } from './elements/SecurityRequirement';
export { default as ServerBindingsElement } from './elements/ServerBindings';
