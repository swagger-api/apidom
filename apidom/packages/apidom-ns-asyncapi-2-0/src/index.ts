import './refractor/registration';

export {
  isRefElement,
  isLinkElement,
  isMemberElement,
  isObjectElement,
  isArrayElement,
  isBooleanElement,
  isNullElement,
  isElement,
  isNumberElement,
  isStringElement,
} from 'apidom';

export { default } from './namespace';

export {
  isAsyncApi2_0Element,
  isAsyncApiVersionElement,
  isChannelBindingsElement,
  isChannelItemElement,
  isChannelsElement,
  isComponentsElement,
  isContactElement,
  isIdentifierElement,
  isInfoElement,
  isLicenseElement,
  isOperationElement,
  isParameterElement,
  isParametersElement,
  isReferenceElement,
  isReferenceElementExternal,
  isSchemaElement,
  isSecurityRequirementElement,
  isServerElement,
  isServerBindingsElement,
  isServersElement,
  isServerVariableElement,
} from './predicates';

export {
  isAsyncApi2_0LikeElement,
  isParameterLikeElement,
  isReferenceLikeElement,
  isSchemaLikeElement,
  isSecurityRequirementLikeElement,
  isServerLikeElement,
  isAsyncApiExtension,
} from './refractor/predicates';

export { keyMap, getNodeType } from './traversal/visitor';

export { default as AsyncApi2_0Element } from './elements/AsyncApi2-0';
export { default as AsyncApiVersionElement } from './elements/AsyncApiVersion';
export { default as ChannelBindingsElement } from './elements/ChannelBindings';
export { default as ChannelItemElement } from './elements/ChannelItem';
export { default as ChannelsElement } from './elements/Channels';
export { default as ComponentsElement } from './elements/Components';
export { default as ContactElement } from './elements/Contact';
export { default as CorrelationIDElement } from './elements/CorrelationID';
export { default as DefaultContentTypeElement } from './elements/DefaultContentType';
export { default as IdentifierElement } from './elements/Identifier';
export { default as InfoElement } from './elements/Info';
export { default as LicenseElement } from './elements/License';
export { default as OperationElement } from './elements/Operation';
export { default as ParameterElement } from './elements/Parameter';
export { default as ParametersElement } from './elements/Parameters';
export { default as ReferenceElement } from './elements/Reference';
export { default as SchemaElement } from './elements/Schema';
export { default as SecurityRequirementElement } from './elements/SecurityRequirement';
export { default as ServerElement } from './elements/Server';
export { default as ServerBindingsElement } from './elements/ServerBindings';
export { default as ServersElement } from './elements/Servers';
export { default as ServerVariableElement } from './elements/ServerVariable';
