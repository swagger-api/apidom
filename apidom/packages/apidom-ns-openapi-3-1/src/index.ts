import './refractor/registration';

export {
  isRefElement,
  isLinkElement,
  isMemberElement,
  isObjectElement,
  isArrayElement,
  isBooleanElement,
  isNullElement,
  isElement,
  isNumberElement,
  isStringElement,
} from 'apidom';

export { default } from './namespace';

export {
  isCallbackElement,
  isComponentsElement,
  isContactElement,
  isExternalDocumentationElement,
  isInfoElement,
  isJsonSchemaDialectElement,
  isLicenseElement,
  isOpenapiElement,
  isOpenApi3_1Element,
  isOperationElement,
  isParameterElement,
  isPathItemElement,
  isPathsElement,
  isReferenceElement,
  isReferenceElementExternal,
  isRequestBodyElement,
  isResponseElement,
  isResponsesElement,
  isSchemaElement,
  isSchemaElementExternal,
  isSecurityRequirementElement,
  isServerElement,
  isServerVariableElement,
  isMediaTypeElement,
} from './predicates';

export {
  isOpenApi3_1LikeElement,
  isParameterLikeElement,
  isReferenceLikeElement,
  isRequestBodyLikeElement,
  isResponseLikeElement,
  isServerLikeElement,
  isOpenApiExtension,
} from './refractor/predicates';

export { keyMap, getNodeType } from './traversal/visitor';

export { default as CallbackElement } from './elements/Callback';
export { default as ComponentsElement } from './elements/Components';
export { default as ContactElement } from './elements/Contact';
export { default as DiscriminatorElement } from './elements/Discriminator';
export { default as ExternalDocumentationElement } from './elements/ExternalDocumentation';
export { default as InfoElement } from './elements/Info';
export { default as JsonSchemaDialectVisitor } from './elements/JsonSchemaDialect';
export { default as LicenseElement } from './elements/License';
export { default as MediaTypeElement } from './elements/MediaType';
export { default as OpenapiElement } from './elements/Openapi';
export { default as OpenApi3_1Element } from './elements/OpenApi3-1';
export { default as OperationElement } from './elements/Operation';
export { default as ParameterElement } from './elements/Parameter';
export { default as PathItemElement } from './elements/PathItem';
export { default as PathsElement } from './elements/Paths';
export { default as ReferenceElement } from './elements/Reference';
export { default as RequestBodyElement } from './elements/RequestBody';
export { default as ResponseElement } from './elements/Response';
export { default as ResponsesElement } from './elements/Responses';
export { default as SchemaElement } from './elements/Schema';
export { default as SecurityRequirementElement } from './elements/SecurityRequirement';
export { default as ServerElement } from './elements/Server';
export { default as ServerVariableElement } from './elements/ServerVariable';
export { default as XmlElement } from './elements/Xml';
