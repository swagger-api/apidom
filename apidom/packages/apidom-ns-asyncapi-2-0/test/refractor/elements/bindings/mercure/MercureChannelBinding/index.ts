import { expect } from 'chai';
import { sexprs } from 'apidom';

import { MercureChannelBindingElement } from '../../../../../../src';

describe('refractor', function () {
  context('elements', function () {
    context('MercureChannelBindingElement', function () {
      specify('should refract to semantic ApiDOM tree', function () {
        const mercureChannelBindingElement = MercureChannelBindingElement.refract({});

        expect(sexprs(mercureChannelBindingElement)).toMatchSnapshot();
      });
    });
  });
});
