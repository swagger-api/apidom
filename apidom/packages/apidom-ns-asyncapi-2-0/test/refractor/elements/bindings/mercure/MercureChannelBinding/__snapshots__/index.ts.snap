// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`refractor elements MercureChannelBindingElement should refract to semantic ApiDOM tree 1`] = `(MercureChannelBindingElement)`;
